### PR TITLE
Add Flex with breakpoint prop

### PR DESCRIPTION
**Pull request by joshri**
Closes: [#2907](https://github.com/weaveworks/weave-gitops-enterprise/issues/2907)

Creates styled component extension of Flex with a breakpoint prop to swap `flex-direction` when below a certain screen width

laggy video here:

https://github.com/weaveworks/weave-gitops/assets/65822698/2fb955b0-feaf-4326-91bb-f0e61578ab06

### CODE REVIEW

**Comment by TheGostKasper** [comment]
I think we don't need to add the breakpoint as we can make use of the `flex-wrap` & `between` for `Flex`


```
   <Flex wide between wrap>
        <SyncActions
          name={name}
          ...
        />
        <Flex gap="14">
          {automation?.type === "HelmRelease" ? (
            ...
          <LargeInfo
           ... />}
          />
        </Flex>
      </Flex>
```

https://github.com/weaveworks/weave-gitops/assets/4614360/0969ea25-34d2-4ce6-b09e-3c24b6835a23

File: ui/components/Flex.tsx [hunk 1]
```diff
@@ -59,4 +59,10 @@ class Flex extends React.PureComponent<Props> {
   }
 }
 
+export const ColumnOnBreakpoint = styled(Styled(Flex))<{ breakpoint: number }>`
```
Comment (jpellizzari) [inline ui/components/Flex.tsx:62]: I dig this :+1:

**Review by jpellizzari — COMMENTED** [review]
I like the creatrivity, but I would also like to see if @TheGostKasper 's suggestion would work here if possible.

**Comment by joshri** [comment]
We'd lose the conditional top padding, which prevents this weird spacing:
image

But also...I might be losing my mind, but when I add the `wrap` prop nothing happens...I have my code in the screenshot to prove I added the prop haha
image

**Comment by TheGostKasper** [comment]
Flex gap controls space in between, you can adjust the right gap. and by adding `wrap` sets whether flex items are forced onto one line or can wrap onto multiple lines.

 [This record](https://github.com/weaveworks/weave-gitops/pull/3953#issuecomment-1683700734)  without setting a gap value.

**Comment by joshri** [comment]
@TheGostKasper Any ideas on why the wrap prop isn't working in my code in the right half of my screenshot?

**Comment by TheGostKasper** [comment]
I believe it's because there's no overlap in width as the Flex Row still show both div without any problems.

**Comment by joshri** [comment]
But the div on the right is clearly off the screen? What am I missing?

**Comment by joshri** [comment]
There was an important diff between EE and OSS here - OSS added a special component for the sync and suspend buttons a couple weeks ago that is now set to 50% width by default, allowing for the large info flex to now wrap when it runs up against it.

**Review by TheGostKasper — APPROVED** [review]

**Pull request merged**